funnel/__init__.py

```diff
@@ -36,13 +37,47 @@
 
 
 # --- Configuration------------------------------------------------------------
+coaster.app.init_app(funnelapp)
 coaster.app.init_app(app)
+
+# These are app specific confguration files that must exist
+# inside the `instance/` directory. Sample config files are
+# provided as example.
+coaster.app.load_config_from_file(funnelapp, 'funnelapp.py')
+coaster.app.load_config_from_file(app, 'hasgeekapp.py')
+
+db.init_app(funnelapp)
 db.init_app(app)
+
+db.app = funnelapp
 db.app = app
+
+migrate = Migrate(funnelapp, db)
 migrate = Migrate(app, db)
+
+mail.init_app(funnelapp)
 mail.init_app(app)
+
+lastuser.init_app(funnelapp)
 lastuser.init_app(app)
```

Comment:
@jace these lines above seems wrong. specially the `db.app = <>` part. How to handle these?

Reply:
`db.app` assignment is only required if there's a database operation outside a request context, to let the db object know which app to operate on. This can happen during init, if we're loading from db, or doing a background job. Workaround: enclose the database operation in `with app.app_context():` (we've previously used `with app.test_request_context():`, a somewhat heavier operation since it also constructs a dummy request).

Reply:
Ok. And what about the others? Double assigning `migrate` and calling `init_app()` multiple times on `db`, `mail` and `lastuser` doesn't seem right.